api/v1alpha1/instancegroup_types.go

```diff
@@ -170,24 +177,34 @@ type EKSSpec struct {
-	KeyPairName                 string              `json:"keyPairName,omitempty"`
-	Image                       string              `json:"image,omitempty"`
-	InstanceType                string              `json:"instanceType,omitempty"`
-	NodeSecurityGroups          []string            `json:"securityGroups,omitempty"`
-	Volumes                     []NodeVolume        `json:"volumes,omitempty"`
-	Subnets                     []string            `json:"subnets,omitempty"`
-	SuspendedProcesses          []string            `json:"suspendProcesses,omitempty"`
-	BootstrapArguments          string              `json:"bootstrapArguments,omitempty"`
-	SpotPrice                   string              `json:"spotPrice,omitempty"`
-	Tags                        []map[string]string `json:"tags,omitempty"`
-	Labels                      map[string]string   `json:"labels,omitempty"`
-	Taints                      []corev1.Taint      `json:"taints,omitempty"`
-	UserData                    []UserDataStage     `json:"userData,omitempty"`
-	ExistingRoleName            string              `json:"roleName,omitempty"`
-	ExistingInstanceProfileName string              `json:"instanceProfileName,omitempty"`
-	ManagedPolicies             []string            `json:"managedPolicies,omitempty"`
-	MetricsCollection           []string            `json:"metricsCollection,omitempty"`
+	EksClusterName              string               `json:"clusterName,omitempty"`
+	KeyPairName                 string               `json:"keyPairName,omitempty"`
+	Image                       string               `json:"image,omitempty"`
+	InstanceType                string               `json:"instanceType,omitempty"`
+	NodeSecurityGroups          []string             `json:"securityGroups,omitempty"`
+	Volumes                     []NodeVolume         `json:"volumes,omitempty"`
+	Subnets                     []string             `json:"subnets,omitempty"`
+	SuspendedProcesses          []string             `json:"suspendProcesses,omitempty"`
+	BootstrapArguments          string               `json:"bootstrapArguments,omitempty"`
+	SpotPrice                   string               `json:"spotPrice,omitempty"`
+	Tags                        []map[string]string  `json:"tags,omitempty"`
+	Labels                      map[string]string    `json:"labels,omitempty"`
+	Taints                      []corev1.Taint       `json:"taints,omitempty"`
+	UserData                    []UserDataStage      `json:"userData,omitempty"`
+	ExistingRoleName            string               `json:"roleName,omitempty"`
+	ExistingInstanceProfileName string               `json:"instanceProfileName,omitempty"`
+	ManagedPolicies             []string             `json:"managedPolicies,omitempty"`
+	MetricsCollection           []string             `json:"metricsCollection,omitempty"`
+	LifecycleHooks              []*LifecycleHookSpec `json:"lifecycleHooks,omitempty"`
```

Comment:
Is a pointer appropriate, do you expect `nil` values in your array?

Reply:
I don't expect nil values, I was using a ptr here because I needed to modify the elements & apply defaults.
I changed it to just make a copy after validation.